### ISSUE TITLE
Question on squeeze(1)

### ISSUE THREAD

**Issue by pone7** [zhihou7/BatchFormer#11]
The shape of the input feature for a transformer is generally (batch, tokens, dim).
As stated in the paper, BatchFormer performs attention at the batch level, but the input shape of this attention layer is (batch, 1, dims) via squeeze operation.
I am considering whether its shape should be like (1, batch, dim)?
Maybe I misunderstood something. 
Looking forward to your reply!

**Comment by zhihou7** [comment]
Hi， thanks for your interest.
It is because the batch dimension in torch.nn. TransformerEncoderLayer is the second dim.

that is (L,B, C)

feel free to comment if you have further questions.

regards,
